src/Microsoft.Data/DataFrame.cs

```diff
@@ -301,116 +352,62 @@ public DataFrame Sort(string columnName, bool ascending = true)
             return new DataFrame(newColumns);
         }
 
-        private void SetSuffixForDuplicatedColumnNames(DataFrame dataFrame, BaseColumn column, string leftSuffix, string rightSuffix)
+        /// <summary>
+        /// Clips values beyond the specified thresholds
+        /// </summary>
+        /// <typeparam name="U"></typeparam>
+        /// <param name="lower">Minimum value. All values below this threshold will be set to it</param>
+        /// <param name="upper">Maximum value. All values above this threshold will be set to it</param>
+        public DataFrame Clip<U>(U lower, U upper)
         {
-            int index = dataFrame._table.GetColumnIndex(column.Name);
-            while (index != -1)
+            DataFrame ret = new DataFrame();
+            for (int i = 0; i < ColumnCount; i++)
             {
-                // Pre-existing column. Change name
-                BaseColumn existingColumn = dataFrame.Column(index);
-                dataFrame._table.SetColumnName(existingColumn, existingColumn.Name + leftSuffix);
-                column.Name += rightSuffix;
-                index = dataFrame._table.GetColumnIndex(column.Name);
+                ret.InsertColumn(0, Column(i).Clip(lower, upper));
```

Comment:
Won't this result in the result data frame having columns in the reversed order?  You are going from 0 .. ColumnCount, and inserting each column at index `0`, which means the last column will be first.

Reply:
Yup, that was a typo. Nice catch!

Reply:
(there should be a whisper in the back of your head, "add a test....")

Reply:
Was there a test/verification added to ensure the columns were in the correct order?

---
In reply to: [313542565](https://github.com/dotnet/corefxlab/pull/2710#discussion_r313542565) [](ancestors = 313542565)